src/features/auth/hooks/useRedirectAfterLogin.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect } from "react";
+
+import qs from "query-string";
+import { useHistory, useLocation } from "react-router-dom";
+
+import { useAppSelector } from "@app/redux/store";
+import { RedirectDef } from "@app/types/route.types";
+
+function useRedirectAfterLogin() {
+  const history = useHistory();
+  const location = useLocation<RedirectDef>();
+  const { redirect } = qs.parse(location.search);
+  const isAuthenticated = useAppSelector(state => state.auth.isAuthenticated);
```

Comment:
Just to match how other files is using `useAppSelector`
```suggestion
  const { isAuthenticated } = useAppSelector(state => state.auth);
```